python/lsst/meas/astrom/matchPessimisticB.py

```diff
@@ -0,0 +1,594 @@
+                                  min([len(refCat), len(goodSourceCat)])))
+
+        # match usable (possibly saturated) sources and then purge saturated
+        # sources from the match list
+        doMatchReturn = self._doMatch(
+            refCat=refCat,
+            sourceCat=goodSourceCat,
+            wcs=wcs,
+            refFluxField=refFluxField,
+            numUsableSources=numUsableSources,
+            minMatchedPairs=minMatchedPairs,
+            match_tolerance=match_tolerance,
+            sourceFluxField=self.sourceSelector.fluxField,
+            verbose=debug.verbose,
+        )
+        matches = doMatchReturn.matches
+        match_tolerance = doMatchReturn.match_tolerance
+
+        if len(matches) == 0:
+            raise RuntimeError("Unable to match sources")
+
+        self.log.info("Matched %d sources" % len(matches))
+        if len(matches) < minMatchedPairs:
+            self.log.warn("Number of matches is smaller than request")
+
+        return pipeBase.Struct(
+            matches=matches,
+            usableSourceCat=goodSourceCat,
+            match_tolerance=match_tolerance,
+        )
+
+    @pipeBase.timeMethod
+    def _doMatch(self, refCat, sourceCat, wcs, refFluxField, numUsableSources,
+                 minMatchedPairs, match_tolerance, sourceFluxField, verbose):
+        """!Implementation of matching sources to position reference stars
+
+        Unlike matchObjectsToSources, this method does not check if the sources
+        are suitable.
+
+        @param[in] refCat  catalog of position reference stars that overlap an
+            exposure
+        @param[in] sourceCat  catalog of sources found on the exposure
+        @param[in] wcs  estimated WCS of exposure
+        @param[in] refFluxField  field of refCat to use for flux
+        @param[in] numUsableSources  number of usable sources (sources with
+            known centroid that are not near the edge, but may be saturated)
+        @param[in] minMatchedPairs  minimum number of matches
+        @param[in] match_tolerance a MatchTolerance object containing
+            variables specifying matcher tolerances and state from possible
+            previous runs.
+        @param[in] sourceInfo  SourceInfo for the sourceCat
+        @param[in] verbose  true to print diagnostic information to std::cout
+
+        @return a list of matches, an instance of
+            lsst.afw.table.ReferenceMatch, a MatchTolerance object
+        """
+
+        # Set configurable defaults when we encounter None type or set
+        # state based on pervious run of matchandfit.
+        if match_tolerance.maxShift is None:
+            maxShiftArcseconds = (self.config.maxOffsetPix *
+                                  wcs.pixelScale().asArcseconds())
+        else:
+            maxShiftArcseconds = np.max(
+                (match_tolerance.maxShift.asArcseconds(),
+                 wcs.pixelScale().asArcseconds()))
```

Comment:
A comment has been added to clarify that, yes, this is a test to keep the max shift allowed from falling below 1 pixel in size.